src/extension.ts

```diff
@@ -164,3 +177,34 @@ function convertLogLevel(commonLogLevel: string) {
             return "FINE";
     }
 }
+
+function resolvePotentialProjectFiles() {
+    let editors = vscode.window.visibleTextEditors.filter((e) => e.document.languageId === "java");
+    const activeEditor = vscode.window.activeTextEditor;
+    if (activeEditor && activeEditor.document.languageId === "java") {
+        const index = editors.indexOf(activeEditor);
+        editors = editors.splice(index, 1);
+        editors = [activeEditor].concat(editors);
+    }
+    if (editors) {
+        const projects = editors.map((e) => {
+            if (e.document.uri.scheme !== "file") {
+                return null;
+            }
+            const workspaceRoot = vscode.workspace.rootPath;
+            const fullpath = e.document.fileName;
+            let cur = path.dirname(fullpath);
+            while (!fs.existsSync(path.join(cur, ".project"))) {
+                if (path.resolve(cur) === workspaceRoot) {
+                    break;
+                }
+                cur = path.join(cur, "../");
```

Comment:
VSCode supports Multi-root workspace. should take care of this case.

Reply:
Besides, if user opens a file that doesn't belong to your workspace, it will go to the endless while loop.

Reply:
nice catch